src/core/definePage.spec.ts

```diff
@@ -254,4 +254,128 @@ export default {
       path: '/custom',
     })
   })
+
+  describe('error handling', () => {
+    it('handles syntax errors gracefully when extracting definePage', async () => {
+      const codeWithSyntaxError = `
```

Comment:
this variable for a piece of code can be extracted in the outer scope. Share it among tests